plugin-infra/go-plugin-infra/src/com/thoughtworks/go/plugin/infra/commons/PluginsZip.java

```diff
@@ -73,7 +74,21 @@ private String computeMd5DigestOfPlugins() {
             String digestOfPlugins = digestOfBundledFolder + digestOfExternalFolder;
             return DigestUtils.md5Hex(digestOfPlugins);
         } catch (Exception e) {
-            throw new RuntimeException(String.format("Could not compute md5 of plugins. Exception occurred: %s", e.getStackTrace()));
+            throw new RuntimeException(String.format("Could not compute md5 of plugins." +
+                    " Exception occurred: %s", e.getStackTrace()));
         }
     }
+
+    private void checkFilesAccessibility(File bundledPlugins, File externalPlugins) {
+        try {
+            if (!bundledPlugins.canRead() || !externalPlugins.canRead()) {
+                LOG.error("Could not read plugins." +
+                        "Please check access rights on files in external and bundled folder.");
```

Comment:
It is ok for lines to be longer than what you have. Feel free to declare this string a single line -

```
                LOG.error("Could not read plugins. Please check access rights on files in external and bundled folder.");
```

A nice improvement would be to print the locations of those folders :)